### PR TITLE
Feature/fix site environment view

### PR DESCRIPTION
**Pull request by bart88**
Fixes issues with with polling on the `site/{id}/environments` page
- Added README on how to develop with this module
- Update package json
- Add jsconfig ( for ides and editors that can read type definitions. see visual studio code )
- Refactor the Table.vue component into more logical functions, fix the interval issues with reconnecting and catch ajax request errors ( also disconnect if hits error limit )

### CODE REVIEW

File: web/modules/custom/shp_vue_view/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Shepherd Vue.js View Module
```
Comment (pingers) [inline web/modules/custom/shp_vue_view/README.md]: This doc should be called DEVELOPERS.md

File: web/modules/custom/shp_vue_view/js/src/table/Table.vue [hunk 2]
```diff
@@ -151,6 +183,33 @@
...
+        },
+        /**
+         * Disconnect the polling interval.
+         *
+         * @param {string} msg - Log message giving reason for disconnection.
+         */
+        disconnectPollingTimer(msg) {
+          clearInterval(poller);
+          poller = null;
+          console.info('Disconnecting polling : ' + msg);
```
Comment (pingers) [inline web/modules/custom/shp_vue_view/js/src/table/Table.vue]: I think the console logging is fine for now - but maybe a future thing would only enable console logging when some debug mode is enabled.

**Comment by pingers** [comment]
Well, I tried it....

I was on the environments page for a site... status went from "Unknown" to "Building". Cool.
I did the site install, but the page never updated to show the new state "Running". We probably give up trying to find the state too quickly - especially when building, which should fairly quickly pass/fail.

File: web/modules/custom/shp_vue_view/js/src/table/Table.vue [hunk 3]
```diff
@@ -151,6 +183,33 @@
...
+          if (reconnectPollerTimer) {
+            clearTimeout(reconnectPollerTimer);
+            reconnectPollerTimer = null;
+          }
+          reconnectPollerTimer = setTimeout(() => {
+            console.info('Restarting polling');
+            pollTiming = POLL_UPDATE_CYCLE;
+            disconnect_count = 0;
+            poller = setInterval(() => { context.poll(); }, pollTiming);
+          }, 900000);
```
Comment (pingers) [inline web/modules/custom/shp_vue_view/js/src/table/Table.vue]: The retries are const's, but not the poll interval?

Comment (bart88) [reply web/modules/custom/shp_vue_view/js/src/table/Table.vue]: The interval is a const but the reconnection timeout wasn't. Changed this.

**Comment by bart88** [comment]
Updated the unchanged state timeout. This should wait a little while longer during a build before disconnecting for 15 minutes.

**Comment by pingers** [comment]
Yeah, this should help. Cheers! :ship:

**Pull request merged**